TombEngine/Scripting/Internal/TEN/Objects/Moveable/MoveableObject.cpp

```diff
@@ -1212,3 +1235,67 @@ void Moveable::AnimFromObject(GAME_OBJECT_ID objectID, int animNumber, int state
 	m_item->Animation.FrameNumber = GetAnimData(*m_item).frameBase;
 	AnimateItem(m_item);
 }
+
+std::unique_ptr<Moveable> Moveable::GetCreatureEnemy()
+{
+	// Check if item is a creature before proceding.
+	if (!this->m_item->IsCreature()) {
+		TENLog(this->GetName() + " is not a creature!\n", LogLevel::Error);
+		return nullptr;
+	}
+	auto creature = GetCreatureInfo(this->m_item);
+	if (!creature->Enemy)
+		return nullptr;
+
+	return std::make_unique<Moveable>(creature->Enemy->Index, true);
+}
+
+void Moveable::ClearCreatureEnemy()
+{
+	// Check if item is a creature before proceding.
+	if (!this->m_item->IsCreature()) {
+		TENLog(this->GetName() + " is not a creature!\n", LogLevel::Error);
+		return;
+	}
+
+	auto creature = GetCreatureInfo(this->m_item);
+
+	creature->Enemy = nullptr;
+}
+
+void Moveable::SetCreatureEnemy(Moveable& enemy)
```

Comment:
I used "enemy" here to make things clear, but I can also do `mov` if preferred.

It can really be any Moveable, creature type or not.